pex/third_party/__init__.py

```diff
@@ -100,9 +100,13 @@ def containing(cls, root):
     prefix = ''
     path = root
     while path:
+      # We use '/' here instead of os.sep because the zip file format spec specifies that paths
+      # must use forward slashes. See section 4.4.17 of
+      # https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT
       if zipfile.is_zipfile(path):
-        return cls(zipfile_path=path, prefix=prefix + os.sep if prefix else '')
-      prefix = os.path.join(prefix, os.path.basename(path))
+        return cls(zipfile_path=path, prefix='{}/'.format(prefix) if prefix else '')
+      path_basename = os.path.basename(path)
+      prefix = '{}/{}'.format(prefix, path_basename) if prefix else path_basename
```

Comment:
What about this:
```python
prefix = os.path.join(prefix, os.path.basename(path)).replace(os.sep, '/')
```

If `os.sep` is already `/`, `replace()` will simply no-op. For example, in a Python REPL `"/".replace("/", "/")` returns back `/`.